src/components/PopupWithForm.js

```diff
@@ -7,9 +7,17 @@ export default class PopupWithFrom extends Popup {
     this._popupForm = this._popupElement.querySelector(".modal__form"); // selects modal form for specific instance of this obj
   }
 
+  setLoading(isLoading, submitSave) {
+    if (isLoading) {
+      this._popupElement.querySelector(".modal__save-button").textContent = "Saving...";
+    } else {
+      this._popupElement.querySelector(".modal__save-button").textContent = submitSave;
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```

Reply:
Try to implement my advice here